build.rs

```diff
@@ -200,6 +200,12 @@ fn main() {
             // As of rustc 1.68, target_feature "lse2" is not available on rustc side:
             // https://github.com/rust-lang/rust/blob/1.68.0/compiler/rustc_codegen_ssa/src/target_features.rs#L58
             target_feature_if("lse2", is_macos, &version, None, false);
+
+            // As of Apple M1/M1 Pro, on Apple hardware, CAS loop-based RMW is much slower than LL/SC
+            // loop-based RMW: https://github.com/taiki-e/portable-atomic/pull/89
+            if is_macos || target_os == "ios" || target_os == "tvos" || target_os == "watchos" {
```

Comment:
This misses Asahi Linux...

Reply:
Is there a way to know that the code is being compiled against Asahi Linux at build time?

Otherwise, it would be reasonable to set `--cfg portable_atomic_ll_sc_rmw` on the user's side.

Reply:
There is an option to revert the whole https://github.com/taiki-e/portable-atomic/commit/93e6ec5f79f6fac70195087020b1a3665505915a, but we want to keep the same codegen as LLVM if possible (LLVM uses CASP regardless of whether it is Apple hardware, though), and we also need to make sure that it does not lose performance on hardware that CASP is fast.

Reply:
> Is there a way to know that the code is being compiled against Asahi Linux at build time?

I can think of is:

- target_vendor. If custom targets are used.
- `-C target-cpu`. If the user uses it.

Both cases can be detected at build time, but I would like to know which one Asahi Linux users actually use.

Reply:
I've implemented the latter approach in https://github.com/taiki-e/portable-atomic/commit/5c3a43b53f1c4188f9dd597599633bc1a315bf44.

```console
$ RUSTFLAGS='-C target-cpu=apple-m1' cargo build -vv --target aarch64-unknown-linux-gnu | grep portable_atomic_ll_sc_rmw
[portable-atomic 1.4.3] cargo:rustc-cfg=portable_atomic_ll_sc_rmw(build)                                                                                                    
```

Reply:
I'm not using any `RUSTFLAGS`, so...

Reply:
To be clear:
- Those specified in .cargo/config (or .cargo/config.toml) are also taken into account.
- If you specify neither target-cpu nor target-feature, the default one (LL/SC) will be used and you should get good performance on apple hardwere.